libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork+Salesforce.m

```diff
@@ -38,20 +40,53 @@ - (void)setDefaultConnectCommunityId:(NSString *)defaultConnectCommunityId {
 
 - (void)setupSalesforceObserver {
     [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(userAccountManagerDidChangeCurrentUser:)
-                                                 name:SFUserAccountManagerDidChangeCurrentUserNotification
+                                             selector:@selector(userAccountManagerDidChangeUserData:)
+                                                 name:SFUserAccountManagerDidChangeUserDataNotification
+                                               object:nil];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(userAccountManagerDidChangeUser:)
+                                                 name:SFUserAccountManagerDidChangeUserNotification
                                                object:nil];
 }
 
 #pragma mark SFAuthenticationManagerDelegate
+- (void)userAccountManagerDidChangeUser:(NSNotification*)notification {
 
-- (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
+    SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
```

Comment:
@khawkins So we should associate the UserAccount to the notification? The Notification is being posted by SFUserAccountManager.